### DISCUSSION TITLE
VIEW / DOWNLOAD terrain with MCC

### DISCUSSION THREAD

**Discussion by Daenges** [ORelio/Minecraft-Console-Client#1826] [Show and tell]
# Introduction
I found a way to view and download terrain with the MCC using [this tool](https://github.com/mircokroon/minecraft-world-downloader).
At the moment, it is more of a workaround, which means that it bugs around, especially on Linux. **Nevertheless** it works on some Servers. I noticed that a ruling factor may be wheather `terrainandmovement` is **supported for the current server version** and activated.

---

# Setup
1. Download both [MCC](https://ci.appveyor.com/project/ORelio/minecraft-console-client/build/artifacts) and [WorldDownloader](https://github.com/mircokroon/minecraft-world-downloader/releases) (Windows-`.exe`/Linux-`.jar`)
2. Make sure `terrainandmovement` is enabled (**MUST BE SUPPORTED** for your version; otherwise it will **not work**.)
3. Start MCC and login (do not connect to any Server) --> `SessionCache.ini` should be created in the local folder

<details>
  <summary>Windows:</summary>

4. Start the downloaded `world-downloader.exe` file
5. Enter the server IP in the `Server adress` box
6. Switch to `Authentication` tab
7. Enter your username
8. Open the `SessionCache.ini`, created in [2]
9. Copy everything from the `=` sign after your E-Mail to the first `,` before your ingame name
10. Paste this into the `Access token` box
11. Click on `Start`

</details>

<details>
  <summary>Linux:</summary>

4. open your terminal
5. Open the `SessionCache.ini`, created in [2]
6. Copy everything from the `=` sign after your E-Mail to the first `,` before your ingame name
7. `java -jar world-downloader.jar -s ip.of.your.server --username IngameName --token TokenFromSessionCache`

</details>

- Switch back to MCC
- connect to `localhost:25565`

--- 

# Result

After this terrain should be rendered in the preview window (e.g. Image). You can now preview and download the world. For more information on this visit the [world-downloader page](https://github.com/mircokroon/minecraft-world-downloader) or their [Wiki](https://github.com/mircokroon/minecraft-world-downloader/wiki).

<details>
  <summary>Show image:</summary>

![image](https://user-images.githubusercontent.com/57369924/142738148-eacc8f50-7210-4ae4-8ab4-fe5f83a85bca.png)

</details>

---

# Disclaimer
Since **nothing of this process is officially supported**, do not expect anything to work. The world-downloader was intentionally designed for the vanilla game client and also works with it flawless.
I experienced **major bugs on Linux** running mono and **long login times** on Windows as well as Linux, while using this method.

--- 

# What are the benefits?
Since I saw some people who wanted to see a **"World-View" feature in MCC**, this could be a pretty easy way to achieve it, if we fix some bugs. Furthermore you can even **download the terrain** with the additional feature that it is currently **not detectable by servers**! You have no "World-Download-Mods" installed which could be detected and banned. You are simply running a **local proxy** which sits between you and the server that logs all incoming world data. (These advantages also apply for the vanilla game client.)

**Comment by ORelio** [comment]
Did not know World Downloader now works as a proxy instead of a mod, this is pretty neat! Thanks for showing this 👍
